### DISCUSSION TITLE
请问图片消息无法序列化怎么办啊

### DISCUSSION THREAD

**Discussion by buliang9988** [simple-robot/Discussions#110] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

在弄一个自定义词库模块，普通消息可以通过序列化的方式存入数据库，但是一旦有图片就无法继续了

**Comment by ForteScarlet** [comment, marked as answer]
消息序列化参考 https://simbot.forte.love/docs/definition/message-overview/message-serialization

如果希望精准控制序列化中具体类型和对应方式，请自行遍历所有消息元素并自行处理

```java
var messages = ...

for (var message : messages) {
    if (message instanceof MiraiImage image) {
        // 如果是mirai图片类型
    } else if (message instanceof Image<?> image) {
        // 其他未知的image类型
    } else if (message instanceof PlainText<?> text) {
        // 文本消息
    } else {
        // 其他...
    }
}
```

**Reply by ForteScarlet** [reply]
如果是词库功能，建议自行约束可用的消息类型和序列化方式然后自行实现转化，这样更可控。
消息序列化更适合用于记录消息内容本身，例如保存消息Log。
